src/dfx/src/lib/builders/motoko.rs

```diff
@@ -166,7 +166,9 @@ impl CanisterBuilder for MotokoBuilder {
         };
         motoko_compile(&self.logger, cache.as_ref(), &params)?;
 
-        if canister_info.get_shrink() {
+        // Default to shrink Motoko canister unless explicitly disable
+        let shrink = canister_info.get_shrink() != Some(false);
```

Comment:
This is better, but still hard to read.  Using `unwrap_or` removes the need for the comment, even.
```suggestion
        let shrink = canister_info.get_shrink().unwrap_or(true);
```